app-emacs/treemacs/treemacs-2.9.ebuild

```diff
@@ -0,0 +1,57 @@
+# Copyright 2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+NEED_EMACS="26.1"
+
+inherit elisp
+
+DESCRIPTION="A tree layout file explorer for Emacs"
+HOMEPAGE="https://github.com/Alexander-Miller/treemacs"
+SRC_URI="https://github.com/Alexander-Miller/treemacs/archive/refs/tags/${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="GPL-3+"
+SLOT="0"
+KEYWORDS="~amd64"
+IUSE="evil magit icons-dired" # some flags need packages not yet in gentoo
+
+RDEPEND=">=app-emacs/ace-window-0.10.0_p20200706
+	>=app-emacs/ht-2.3
+	>=app-emacs/hydra-0.15.0
+	>=app-emacs/pfuture-1.9
+	>=app-emacs/cfrs-1.3.2
+	evil? ( app-emacs/evil )
+	magit? ( app-emacs/magit )"
+DEPEND="${RDEPEND}"
+
+RESTRICT="test" # test requires cask and buttercup.el which are not packaged
+SITEFILE="50treemacs-gentoo.el"
+DOCS=( README.org Changelog.org Extensions.org )
+
+src_unpack() {
+	elisp_src_unpack
+	mv "${S}"/src/{elisp,scripts}/* "${S}/" || die
+	if use evil; then
+		mv "${S}/src/extra/treemacs-evil.el" "${S}/" || die
+	fi
+	if use magit; then
+		mv "${S}/src/extra/treemacs-magit.el" "${S}/" || die
+	fi
+	if use icons-dired; then
+		mv "${S}/src/extra/treemacs-icons-dired.el" "${S}/" || die
+	fi
+}
+
+src_compile() {
+	elisp-make-autoload-file "${S}/${PN}-autoload.el" "${S}/"
+	elisp_src_compile
+}
+
+src_install() {
+	elisp_src_install
+	mkdir -p "${D}/usr/share/${PN}/icons/" || die
+	cp -r "${S}/icons/default/" "${D}/usr/share/${PN}/icons/" || die
+	ln -s "/usr/share/${PN}/icons/" "${D}/${SITELISP}/${PN}/icons" || die
+	cp "${S}"/*.py "${D}/${SITELISP}/${PN}/" || die
```

Comment:
doins and ${ED} if you were going to stuck with cp

Reply:
Also, *.py makes me wonder if we need to use a Python eclass..?

Reply:
There are only 3 python files in src/scripts, and treemacs searches for in the directory that the elisp files get dumped in. Do you agree that a python eclass is overkill?

Byte compiling the python scripts also probably won't work because it searches for files with a .py in the directory it's installed into, but I'm not sure.

Reply:
I think [python-single-r1](https://dev.gentoo.org/~mgorny/python-guide/single.html) would be right here: if we're installing Python files, they might stop working with a future Python version, or need a minimum Python version => we should declare PYTHON_COMPAT.

Also, using a Python eclass means we can fix the shebang on the files to match the Python interpreter the user chose to install this with (via PYTHON_SINGLE_TARGET) rather than running whatever 'python' happens to be on the system.

Agree it feels like overkill but I think it's needed here?